crates/nargo/tests/test_data/strings/src/main.nr

```diff
@@ -0,0 +1,8 @@
+fn main(message : pub [char; 11]) {
```

Comment:
I tried to have the syntax be `str<N>` but I kept running into this build error: 
```
error: Unexpected end of input, expected one of >, [, binary operator, cast, field access
  ┌─ /Users/maximvezenov/Documents/dev/noir/crates/nargo/tests/test_data/strings/src/main.nr:8:1
  │
8 │ }
  │ -
error: aborting due to 1 previous errors
```
I ended up going with this syntax for now `str[N]`. Do you have an idea of what might be causing this? The parser is the exact same except the tokens delimiting the expression for string size were `Token::Less` and `Token::Greater`.

Reply:
I don't like the `str[N]` syntax, as it look like we have an array of strings.
So I would go for `str<N>`, or only `char[N]` until we get the `str<N>` working.
Also, why not using `string` rather than `str`?

Reply:
We can change to `str<N>` once #620 is merged since that provides support for parsing numeric generics within regular brackets. Previously it was an ambiguous parse as we tried to parse arbitrary expressions where the terminating `>` is also a valid operator.
I'm guessing `str` was chosen over `string` here since rust also uses the `str` keyword for its string reference type `&str`.